### PR TITLE
feat: add development guideline file

### PR DESCRIPTION
**Pull request by mohnoor94**
# Situation
The project lacked a comprehensive development guide, which made it difficult for new contributors to get started with the project.

# Task
The goal is to create a development guide that outlines the necessary steps for developers to set up the project, build the SDK, and address common issues they may encounter.

# Action
1. Created a `DEVELOPMENT.md` file in the project's root directory.
2. Included the following sections:
   - Development Requirements
   - Quick Start
     - Cloning the repository
     - Initializing the GraphQL schema
     - Building the project
   - Project Structure
   - Common Issues
     - Gradle JDK detection
     - Build failures
   - Questions and Issues

# Testing
The development guide was tested by following the instructions and verifying that the project could be successfully set up and built on a local machine.

# Results
The new development guide provides clear and concise instructions for developers to get started with the project. It covers the essential steps, common troubleshooting, and a high-level overview of the project structure.

# Notes
- Internal ticket: SDK-1570

### CODE REVIEW

**Review by dtayeh — DISMISSED** [review]

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+## Quick Start
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/ExpediaGroup/lodging-connectivity-java-sdk.git
+cd lodging-connectivity-java-sdk
+```
+
+### 2. Initialize GraphQL Schema
```
Comment (Mohammad-Dwairi) [inline DEVELOPMENT.md:20]:
```suggestion
### 2. Initialize GraphQL Module
```
It's not only the schema, we have operations, fragments, and schema files

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+│   ├── graphql/         # GraphQL schema and queries (Sub module)
+│   └── kotlin/          # Kotlin source code of the SDK
+├── examples/
+│   └── src/main/java/   # Example usage of the SDK in Java
+├── docs/                # Documentation files
+├── scripts/
+│   └── graphql/         # GraphQL setup scripts
+├── apollo-compiler-plugin/
+│   └── src/main/        # Apollo compiler plugin source code
+├── build.gradle.kts     # Gradle build configuration
```
Comment (Mohammad-Dwairi) [inline DEVELOPMENT.md:56]:
```suggestion
├── build.gradle    # Gradle build configuration
```

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+├── gradle.properties    # Gradle properties
+├── settings.gradle.kts  # Gradle settings
```
Comment (Mohammad-Dwairi) [inline DEVELOPMENT.md:58]:
```suggestion
├── settings.gradle # Gradle settings
```

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+└── ...
+```
+
+## Common Issues
+
+### 1. Gradle Cannot Detect JDK
+
+If Gradle fails to detect your JDK installation, try either:
+
+Option 1: Add this line to your `gradle.properties`:
```
Comment (Mohammad-Dwairi) [inline DEVELOPMENT.md:68]:
```suggestion
Option 1: Add this line to the root `gradle.properties`:
```

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+```bash
+export JAVA_HOME=$(/usr/libexec/java_home -v 11)
+```
+
+### 2. Build Failures
+
+- Ensure Java 11 is installed and configured as your development JDK
+- Check your Java version: `java -version`
+- Make sure you've run the GraphQL schema initialization script
```
Comment (Mohammad-Dwairi) [inline DEVELOPMENT.md:84]:
```suggestion
- Make sure you've run the GraphQL module initialization script
```

**Review by Mohammad-Dwairi — APPROVED** [review]

**Review by jordan-n-schmidt — APPROVED** [review]

**Pull request merged**